docs/src/design/proposals/APIRedesign_PeteRichards.md

```diff
@@ -0,0 +1,251 @@
+- [Reducing interface depth (The angular discussion)](#reducing-interface-depth-the-angular-discussion)
+  - [More angular: for all services](#more-angular-for-all-services)
+  - [Less angular: only for views](#less-angular-only-for-views)
+- [Standard packaging and build system](#standard-packaging-and-build-system)
+  - [Use systemjs for module loading](#use-systemjs-for-module-loading)
+  - [Use gulp or grunt for standard tooling](#use-gulp-or-grunt-for-standard-tooling)
+  - [Package openmctweb as single versioned file.](#package-openmctweb-as-single-versioned-file)
+- [Misc Improvements](#misc-improvements)
+  - [Refresh on navigation](#refresh-on-navigation)
+  - [Move persistence adapter to promise rejection.](#move-persistence-adapter-to-promise-rejection)
+  - [Remove bulk requests from providers](#remove-bulk-requests-from-providers)
+- [Notes on current API proposals:](#notes-on-current-api-proposals)
+- [[1] Footnote: The angular debacle](#1-footnote-the-angular-debacle)
+  - ["Do or do not, there is no try"](#do-or-do-not-there-is-no-try)
+  - [A lack of commitment](#a-lack-of-commitment)
+  - [Commitment is good!](#commitment-is-good)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+# Reducing interface depth (the bundle.json version)
+
+## Imperitive component registries 
+
+Transition component registries to javascript, get rid of bundle.json and bundles.json.  Prescribe a method for application configuration, but allow flexibility in how application configuration is defined.
+
+Register components in an imperitive fashion, see angularApp.factory, angularApp.controller, etc.  Alternatively, implement our own application object with new registries and it's own form of registering objects.
+
+## Get rid of "extension category" concept.
+
+The concept of an "extension category" is itself an extraneous concept-- an extra layer of interface depth, an extra thing to learn before you can say "hello world".  Extension points should be clearly supported and documented with whatever interfaces make sense.  Developers who wish to add something that is conceptually equivalent to an extension category can do so directly, in the manner that suites their needs, without us forcing a common method on them.
+
+## Reduce number and depth of extension points
+
+Clearly specify supported extension points (e.g. persistence, model providers, telemetry providers, routes, time systems), but don't claim that the system has a clear and perfect repeatable solution for unknown extension types.  New extension categories can be implemented in whatever way makes sense, without prescribing "the one and only system for managing extensions".
+
+The underlying problem here is we are predicting needs for extension points where none exist-- if we try and design the extension system before we know how it is used, we design the wrong thing and have to rewrite it later.
```

Comment:
Not sure I understand the practical implications here.

I do think "extensible by default" should continue to be a fundamental part of our approach. Not saying we should carry a lot of unused extension points, but when we introduce new functionality we should prefer to do so with extensions over local code modifications, even if it means introducing a new extension point. This has multiple benefits:

* Avoids accumulation of complexity in individual units; baseline behavior is a separate concern from enhancements.
* Increases opportunity for adoption (want to use Open MCT Web but it doesn't do X? there's a reasonable chance there's an extension you can leverage, and that chance is always increasing.)
* Improves our ability to respond to evolving and/or diverging requirements (add/remove extensions)

I do think reducing the number of extension points is good when there is no loss to extensibility (e.g. consolidating views/representations/templates into one type, shifting the distinguishing characteristics elsewhere), as this reduces the number of interfaces.